pipeline/00-ingest.R

```diff
@@ -205,21 +206,57 @@ message("Adding time features and cleaning")
 # parking spot, but only the sale for the unit, not the parking. Drop other
 # multi-unit sale types since we don't have a way to disaggregate each
 # unit's value
-training_data <- training_data %>%
+training_data_ms <- training_data %>%
   group_by(meta_sale_document_num) %>%
-  arrange(meta_tieback_proration_rate) %>%
+  arrange(meta_sale_document_num, meta_tieback_proration_rate) %>%
   mutate(
+    # Attach sale to the condo UNIT if one of the PINs in the sale is a garage
+    # and the unit % of ownership is greater than 3x the garage % of ownership
     keep_unit_sale =
-      meta_tieback_proration_rate >= (lag(meta_tieback_proration_rate) * 3)
+      meta_tieback_proration_rate >= (lag(meta_tieback_proration_rate) * 3) &
+        sum(meta_cdu == "GR", na.rm = TRUE) == 1, # nolint
+    # If there are multiple PINs associated with a sale, take only the
+    # proportion of the sale value that is attributable to the main unit (based
+    # on percentage of ownership)
+    total_proration_rate = sum(meta_tieback_proration_rate, na.rm = TRUE),
+    meta_sale_price = as.numeric(meta_sale_price),
+    meta_sale_price = ifelse(
+      n() == 2 & keep_unit_sale,
+      meta_sale_price * (meta_tieback_proration_rate / total_proration_rate),
+      meta_sale_price
+    ),
```

Comment:
This is the most critical change in this PR, as its directly adjusting the outcome. Here we're trying to pull out the proportion of the total sale price attached to _only the unit_.

Reply:
Could you just add a note that `sum(meta_cdu == "GR", na.rm = TRUE) == 1` is how we make sure 1 (and only 1) of the parcels being sold in a multisale is a parking space?

Reply:
Clarified in 6d8055d